src/core/copyright/copyright_dpkgquery.cpp

```diff
@@ -10,7 +10,13 @@ namespace linuxdeploy {
             using namespace log;
 
             std::vector<fs::path> DpkgQueryCopyrightFilesManager::getCopyrightFilesForPath(const fs::path& path) {
-                subprocess::subprocess proc{{"dpkg-query", "-S", path.c_str()}};
+                fs::path realpath = fs::canonical(path);
+
+                if ((std::string)realpath != (std::string)path) {
```

Comment:
No C-style casts in C++. Please just use the constructor.

Reply:
thank you i didn't know i could do that. i don't write much c++ and didn't know casting isn't equivalent to construction.